src/mpid/ch4/netmod/ofi/ofi_win.h

```diff
@@ -1062,6 +1068,8 @@ static inline int MPIDI_NM_mpi_win_allocate_shared(MPI_Aint size,
         shm_unlink(shm_key);
 
   fn_exit:
+    if (shm_key != NULL)
```

Comment:
Ah, yes I see the concern. But still, if you have `NULL` initialized the `shm_key` pointer, it should be safe to pass to `free`, right?